cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/prometheus/InvalidPrometheusResultException.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.monitor.sampling.prometheus;
+
+import com.linkedin.kafka.cruisecontrol.exception.KafkaCruiseControlException;
+
+/**
+ * The exception indicates that the broker with which the metric is associated could not be determined.
+ */
+public class InvalidPrometheusResultException extends KafkaCruiseControlException {
```

Comment:
Multiple Kafka clusters can use the same Prometheus server to publish metrics. In such a case, the Prometheus API queries may return metrics from hosts that are not part of the cluster being managed by Cruise Control. When we receive such query results, we simply want to ignore them and move on to the other query results. We do not want to abandon the entire sampling run because some of the query results were invalid.

The current usage of the `SamplingException` seems to be in cases where we want to abandon the current sampling run because of an irrecoverable error.

To distinguish from these errors, we have created `InvalidPrometheusResultException`, which are not supposed to abandon the sampling execution, but simply skip the current metric result being evaluated because it is invalid.